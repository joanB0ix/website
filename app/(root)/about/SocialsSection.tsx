import { Section } from "@/components/Section";
import { Social } from "@/config/socialList";
import Link from "next/link";

export interface SocialsSectionProps {
  socialList: Social[];
}

export default function SocialsSection({ socialList }: SocialsSectionProps) {
  return (
    <Section title="My Socials">
      <ul className="list-disc pl-6">
        {socialList.map((social) => {
          return (
            <li className="underline" key={social.name}>
              <Link href={social.url}>{social.name}</Link>
            </li>
          );
        })}
      </ul>
    </Section>
  );
}
