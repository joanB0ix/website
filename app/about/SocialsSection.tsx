import { Social } from "@/config/socialList";
import Link from "next/link";
import { Section } from "../../components/Section";

export interface SocialsSectionProps {
  socialList: Social[];
}

export default function SocialsSection({ socialList }: SocialsSectionProps) {
  return (
    <Section title="My Socials">
      <ul className="list-disc pl-6">
        {socialList.map((social) => {
          return (
            <li className="underline" key={social.name}>
              <Link href={social.url}>{social.name}</Link>
            </li>
          );
        })}
      </ul>
    </Section>
  );
}
