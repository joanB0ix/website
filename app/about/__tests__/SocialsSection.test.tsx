import { render, screen } from "@testing-library/react";
import SocialsSection from "../SocialsSection";

jest.mock("next/link", () => {
  const Link = ({
    href,
    children,
  }: {
    href: string;
    children: React.ReactNode;
  }) => <a href={href}>{children}</a>;
  Link.displayName = "MockedLink";
  return Link;
});

const SOCIAL_LIST = [
  { name: "GitHub", url: "https://github.com/example" },
  { name: "LinkedIn", url: "https://linkedin.com/in/example" },
];

describe("SocialsSection", () => {
  it("Should render correctly", () => {
    render(<SocialsSection socialList={SOCIAL_LIST} />);

    const sectionTitle = screen.getByText("My Socials");
    expect(sectionTitle).toBeInTheDocument();

    SOCIAL_LIST.forEach((social) => {
      const socialLink = screen.getByText(social.name);
      expect(socialLink).toBeInTheDocument();
      expect(socialLink.closest("a")).toHaveAttribute("href", social.url);
    });
  });

  it("Should not render list when there are no socials", () => {
    render(<SocialsSection socialList={[]} />);
    const listItems = screen.queryAllByRole("listitem");
    expect(listItems).toHaveLength(0);
  });

  it("Should match snapshot", () => {
    const { asFragment } = render(<SocialsSection socialList={SOCIAL_LIST} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
