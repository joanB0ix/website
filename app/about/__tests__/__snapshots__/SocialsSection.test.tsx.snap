// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialsSection Should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="my-6"
  >
    <h2
      class="font-bold pb-2"
    >
      My Socials
    </h2>
    <ul
      class="list-disc pl-6"
    >
      <li
        class="underline"
      >
        <a
          href="https://github.com/example"
        >
          GitHub
        </a>
      </li>
      <li
        class="underline"
      >
        <a
          href="https://linkedin.com/in/example"
        >
          LinkedIn
        </a>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
